src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,773 @@
+                operation => operation.ContextualPath,
+                operation => operation);
+            return new SortedDictionary<RequestPath, MgmtRestOperation>(operationMappings, Comparer<RequestPath>.Create(
+                (x1, x2) =>
+                {
+                    // order by path length descendently
+                    if (x1.ToString() is null)
+                    {
+                        return 1;
+                    }
+                    else if (x2.ToString() is null)
+                    {
+                        return -1;
+                    }
+
+                    return x2.ToString()!.Length - x1.ToString()!.Length;
+                }));
+        }
+
+        public static bool HasExample(BuildContext<MgmtOutputLibrary> context, MgmtClientOperation? clientOperation)
+        {
+            if (clientOperation is null)
+            {
+                return false;
+            }
+
+            foreach ((var branch, var operation) in GetSortedOperationMappings(clientOperation))
+            {
+                var exampleGroup = MgmtBaseTestWriter.FindExampleGroup(context, operation);
+                if (exampleGroup is not null && exampleGroup.Examples.Count > 0)
+                    return true;
+                break;
+            }
+            return false;
+        }
+
+        public static ObjectTypeProperty? FindPropertyThroughSerialization(List<string> flattenedNames, JsonSerialization js)
+        {
+            if (flattenedNames.Count == 0)
+            {
+                return null;
+            }
+            if (js is JsonObjectSerialization obj)
+            {
+                foreach (JsonPropertySerialization property in obj.Properties)
+                {
+                    if (flattenedNames[0] == property.Name)
+                    {
+                        if (flattenedNames.Count == 1)
+                            return property.Property;
+                        return FindPropertyThroughSerialization(flattenedNames.Skip(1).ToList(), property.ValueSerialization);
+                    }
+                }
+            }
+            return null;
+        }
+
+        public static ExampleValue? FindPropertyValue(ObjectType ot, ExampleValue ev, ObjectTypeProperty targetProperty)
+        {
+            if (ot is SchemaObjectType sot)
+            {
+                var obj = (JsonSerialization)sot.Serializations.Single(x => typeof(JsonSerialization).IsInstanceOfType(x));
+                foreach (var exampleProperty in ev.Properties ?? new DictionaryOfExamplValue())
+                {
+                    var property = FindPropertyThroughSerialization(exampleProperty.Value.FlattenedNames is not null ? exampleProperty.Value.FlattenedNames.ToList()! : new List<string> { exampleProperty.Key }, obj);
+                    if (property == targetProperty)
+                    {
+                        return exampleProperty.Value;
+                    }
+                }
```

Comment:
These two methods look quite low-level. In a writer, I believe we should not do anything low-level. The writer should only write things without calculating anything.
Do we have a better place for these two methods?